### DISCUSSION TITLE
Metronome as a standard element of the game

### DISCUSSION THREAD

**Discussion by Moltanicaa** [ppy/osu#13129] [Ideas]
**Describe the new feature:** 
When using the NC mod, we hear extra bits that work like a metronome. Maybe it’s worth adding the ability to turn on/off the metronome for NC and other modes? It’s more convenient for someone to play without these “duts-duts-duts”, and for some it helps to keep the rhythm.
**Proposal designs of the feature:** 
Let it work the same way as for NC.

**Comment by peppy** [comment]
I'd argue this takes away any kind of music-sourced difficulty. Not sure that's something we want.

**Comment by gagahpangeran** [comment]
This can be some kind of mod, like metronome mod when you can turn on and off.

Player can use this mod to exercise keep the rhythm and new player can use this mod so they can 'hear' the beat more clearly. Add custom option for metronome tick every 1, 1/2, or 1/4 beat.

**Comment by Moltanicaa** [comment]
Today we discussed this topic in the russian community and many spoke out that we would like to play without extra sounds with NC. Already, you can simply remove the sounds of this mod from the skin, so this would be able to follow the rhythm of a large number of people, because not everyone has a developed musical ear. 

In any case, the loud sound “tuts-tuts-tuts” does not allow you to play NC with comfort, because of it you have to lower the volume of hitsound, which makes the gameplay not quite comfortable. Maybe then make a separate volume for him? Although this is completely stupid.

**Comment by LastExceed** [comment]
> not everyone has a developed musical ear

if you can't stay on beat or understand complex rythms then you will score poorly. That is the very essence of a rythm game. Adding a metronome to compensate that is like adding an auto-clicker to compensate for one's inability to stream fast, it completely defeats the point of having a learning curve in the game

> you can use NC as a metronome in stable already

that falls into the topic of skin-cheating, which is a separate discussion that has yet to be concluded

I do think metronome would make a nice addition to the game as a mod, but it should have a score penalty or be unranked.

**Comment by Moltanicaa** [comment]
Sounds fair. Then the mod option looks more interesting.

**Comment by sugoku** [comment]
This is an older issue but I'd like to bring up a different perspective. I have been playing osu! while trying to avoid piracy/unlicensed content; that is, I have been muting songs which are unlicensed in osu! or that I cannot legally play.

I often use NC to keep my rhythm without music and sometimes I will create a beat pattern or click track to replace the audio of a song entirely if it is that much harder to keep rhythm without the original audio playing. I have been playing HDHR (OD10) often without music and it has caused problems especially with breaks and slower songs.

I would like to argue that a default instrumental track should be allowed without penalty to score or pp. It doesn't have to be a click track or a metronome; it can be similar or the same as NC behavior with a drum pattern that can at least keep you in rhythm. However, if it becomes an option or a mod, I think the actual audio track should be muted entirely, in the case that the track may provide unfair advantages (perhaps with polyrhythms or other problematic rhythms).

The other alternative I'd like to suggest is allowing a metronome to be enabled during song breaks. One of the most frequent issues I have is keeping track of the rhythm in between breaks when the song is muted, so that would solve a large part of the issue.

In any case, I think this would help to ensure that osu! can always be played legally when needed.
For example, this would be useful when streaming to avoid copyright issues.

Please let me know of any thoughts you have; I would be willing to help with the pull request.

**Comment by bdach** [comment]
Sorry but I find the notion of playing a rhythm game without music absolutely baffling and don't think that it should be a justification for this feature (or any feature). The featured artist program is an actual solution for your "licensing" woes.

Additionally, it is already possible to mute the audio track while leaving sound effects on.

**Comment by sugoku** [comment]
I appreciate the feedback. Unfortunately although I believe the Featured Artist program is an amazing step in the right direction it does not encompass the majority of maps in the game, and this can be problematic especially in competitive play when trying to get top scores. 

My intention is to still have a musical rhythm in the background so that I don't have to solely rely on my own internal rhythm, which despite being an interesting challenge in itself, is not a replacement for actual music when playing, as you said. Muting the audio track while leaving sound effects on is generally my current strategy but sliderends and approach circles are the only things which help to guide internal rhythm. That's my motive for a possible "fallback" metronome feature or a metronome mod in general, and I respect your opinion if you think it is absurd.

**Comment by peppy** [comment]
Going to lock this thread for now as I don't think this is going to be something we immediately add (and there's although enough opinions as to why / why not this should be implemented).

As always, if this is something you would like to see in the game user the reaction option on the first post.
